### DISCUSSION TITLE
[DOM] Example of skewing arrow element

### DISCUSSION THREAD

**Discussion by lbernick** [floating-ui/floating-ui#2801] [Q&A]
Hi, I saw in the docs for the arrow element that [centerOffset can be used to skew an arrow](https://floating-ui.com/docs/arrow#centeroffset) so it stays pointing towards the reference element. I'm using padding for both the `shift` and `arrow` middleware, and my arrow no longer points to the reference element:
<img width="158" alt="Screenshot 2024-02-21 at 11 08 09 AM" src="https://github.com/floating-ui/floating-ui/assets/29333301/7da2ea2d-0f14-4c89-9f00-21ea374df7f0">

Do you have any examples of how to calculate the correct skew? I found https://github.com/floating-ui/floating-ui/pull/2423 which looks like exactly what I want but I'm not quite sure how to adapt that code for vanilla js.

Code snippet:
```
    const arrowLen = arrow.offsetWidth
    const hypotenuse = Math.sqrt(2 * arrowLen ** 2)
    const floatingOffset = hypotenuse / 2

    computePosition(referenceElement, tooltip, {
      middleware: [
        offset(floatingOffset),
        flip(),
        shift({ padding: 10 }),
        floatingArrow((state) => ({ element: arrow, padding: state.rects.reference.width / 2 })),
      ],
    }).then(({ x, y, middlewareData, placement }) => {
      Object.assign(tooltip.style, {
        left: `${x}px`,
        top: `${y}px`,
      })
      const side = placement.split("-")[0]
      let staticSide = oppositeSide[side]
      if (middlewareData.arrow) {
        const { x: arrowX, y: arrowY } = middlewareData.arrow
        Object.assign(arrow.style, {
          left: arrowX != null ? `${arrowX}px` : "",
          top: arrowY != null ? `${arrowY}px` : "",
          transform: `rotate(45deg)`,
          [staticSide]: `${-arrowLen / 2}px`,
        })
      }
    })
```

Thanks for your help!

**Comment by atomiks** [comment]
The math used only works on an SVG element, not a plain rotated HTML element. The code will work for vanilla JS assuming you switch to an SVG arrow.
